### PR TITLE
fix consensus stop when remove leader

### PR DESCRIPTION
**Pull request by Jayanring**

### CODE REVIEW

File: src/peer.rs [hunk 1]
```diff
@@ -420,12 +417,63 @@ impl Peer {
...
             }
 
+            //if incoming change need to transfer leader
+            if self.is_leader() && !self.pending_conf_change.is_empty() {
+                let incoming_conf_change = self.pending_conf_change.last().unwrap();
+                if incoming_conf_change.changes.iter().any(|c| {
+                    c.change_type == ConfChangeType::RemoveNode as i32
+                        && c.node_id == self.core.raft.id
+                }) {
+                    self.need_to_transfer_leader = true;
```
Comment (NaughtyDogOfSchrodinger) [inline src/peer.rs]: need_to_transfer_leader使用局部变量判断即可

File: src/peer.rs [hunk 2]
```diff
@@ -679,35 +729,53 @@ impl Peer {
...
+                warn!(self.logger, "all validators replaced");
+                let conf_change_add = ConfChangeV2 {
+                    changes: added_ccs,
+                    ..Default::default()
+                };
+                let conf_change_remove = ConfChangeV2 {
+                    changes: removed_ccs,
+                    ..Default::default()
+                };
+                self.pending_conf_change = vec![conf_change_remove, conf_change_add];
```
Comment (NaughtyDogOfSchrodinger) [inline src/peer.rs:785]: 最好加些注释增加可读性

File: src/peer.rs [hunk 3]
```diff
@@ -420,12 +417,90 @@ impl Peer {
...
+                        let current_peers: HashSet<u64> = self
+                            .core
+                            .store()
+                            .get_conf_state()
+                            .voters
+                            .iter()
+                            .copied()
+                            .collect();
+                        //AddNode = 0; RemoveNode = 1;
+                        match c.change_type {
```
Comment (NaughtyDogOfSchrodinger) [inline src/peer.rs]: pending_conf_change长度为2，pending_conf_change[1]类型全是Added

**Comment by Jayanring** [comment]
close #57

**Comment by Jayanring** [comment]
[get more details](https://bn79jc9x8i.feishu.cn/file/boxcnuFBG94W8lxcqs6qWvZoLXQ)

**Review by JLerxky — APPROVED** [review]

**Review by NaughtyDogOfSchrodinger — APPROVED** [review]

**Pull request merged**